docker/opbeans/rum/tasks.js

```diff
@@ -10,30 +10,34 @@ function sleep(ms) {
 }
 
 async function run() {
-    const chromeless = new Chromeless({
-        lauchChrome: true
-    })
+    try {
+        const chromeless = new Chromeless({
+            lauchChrome: true
+        })
 
-    for (; ;) {
-        url = await chromeless
-            .goto(url)
-            .evaluate((baseUrl) => {
-                // activateLoadGeneration is defined in opbeans-frontend
-                if (typeof window.activateLoadGeneration === 'function') {
-                    console.log('Activating route change load generation')
-                    window.activateLoadGeneration()
-                }
+        for (; ;) {
+            url = await chromeless
+                .goto(url)
+                .evaluate((baseUrl) => {
+                    // activateLoadGeneration is defined in opbeans-frontend
+                    if (typeof window.activateLoadGeneration === 'function') {
+                        console.log('Activating route change load generation')
+                        window.activateLoadGeneration()
+                    }
 
-                var links = document.querySelectorAll('a[href^="/"]')
-                if (links && links.length) {
-                    var i = Math.floor(Math.random() * links.length)
-                    return links[i].href
-                } else {
-                    return baseUrl
-                }
-            }, baseUrl);
-        console.log(url)
-        await sleep(8000 + Math.floor(Math.random() * 10000))
+                    var links = document.querySelectorAll('a[href^="/"]')
+                    if (links && links.length) {
+                        var i = Math.floor(Math.random() * links.length)
+                        return links[i].href
+                    } else {
+                        return baseUrl
+                    }
+                }, baseUrl);
+            console.log(url)
+            await sleep(8000 + Math.floor(Math.random() * 10000))
+        }
+    } catch (e){
```

Comment:
if the RUM agent is nor running for some reason, it would be stuck forever

Reply:
or when the stack is full

Reply:
If that's the case the timeout in the pipeline should kill it, not ideal, neither I know whether the use case of running the opbeans-rum without the RUM agent might be something which might happen.

Reply:
I bet the time out is related to this line https://github.com/elastic/opbeans-frontend/blob/980651d5ca43124a686e9b1fd0e33483173375e5/src/components/ProductDetail/index.js#L8-L10

Reply:
I workaround this random fail in the Docker image by adding this config to the RUM agent https://github.com/elastic/apm-integration-testing/blob/master/docker/opbeans/frontend_nginx/rum-config.template

Reply:
Most of the opbeans apps add random failures to simulate real applications. 
I could be wrong but the random error seen here should only cause an uncaught exception.   It shouldn't cause Chrome to timeout like we see.

afaict the fix by @v1v works. You _could_ add a guard like:
```ts
} catch (e) {
  if (e.message.includes("timed out after")) {
    return run();
  }
  throw e;
}
```